runtime/compiler/control/MongoLogger.cpp

```diff
@@ -0,0 +1,161 @@
+#include "MongoLogger.hpp"
+#include <ctime>
+#include <stdio.h>
+#include <string.h>
+#include <string>
+#include <stdlib.h>
+#include "LoadDBLibs.hpp"
+
+MongoLogger::MongoLogger(const char* databaseIP, uint32_t databasePort, const char* databaseName)
+      : BasePersistentLogger(databaseIP, databasePort, databaseName)
+   {
+   init();
+   }
+
+MongoLogger::MongoLogger(const char* databaseIP, uint32_t databasePort, const char* databaseName,
+                         const char* databaseUsername, const char* databasePassword)
+      : BasePersistentLogger(databaseIP, databasePort, databaseName, databaseUsername, databasePassword)
+   {
+   init();
+   }
+
+MongoLogger::~MongoLogger()
+   {
+   //Clean up this mongoc logger.
+   Omongoc_collection_destroy(_collection);
+   Omongoc_database_destroy(_db);
+   Omongoc_client_destroy(_client);
+   Omongoc_uri_destroy(_uri);
+   }
+
+void MongoLogger::init()
+   {
+   _uri = NULL;
+   _collection = NULL;
+   _db = NULL;
+   _client = NULL;
+   }
+
+char * MongoLogger::constructURI()
+   {
+   // Check if we have the database name
+   if (strcmp(_databaseName,"") == 0)
+      {
+      return "jitserver_logs";
+      }
+
+   // Check if we have db IP and Port
+   if (strcmp(_databaseIP,"") == 0)
+      {
+      // No IP try localhost
+      _databaseIP = "127.0.0.1";
+      }
+   if (!_databasePort)
+      {
+      // No Port try default MongoDB port
+      _databasePort = 27017;
+      }
+
+   // Check if we have credentials
+   if (strcmp(_databaseUsername,"") != 0)
+      {
+      if (strcmp(_databasePassword,"") != 0)
+         {
+         snprintf(_uri_string, 256, "mongodb://%s:%s@%s:%u/?authSource=%s", _databaseUsername, _databasePassword,
```

Comment:
It's not apparent were this 256 is coming from. Let's use some symbolic name.
Please check for buffe overflow.